src/common/test-channel.c

```diff
@@ -469,6 +469,83 @@ test_capable (void)
   g_object_unref (transport);
 }
 
+static void
+test_ping_channel (void)
+{
+  JsonObject *reply = NULL;
+  JsonObject *options;
+  MockTransport *mock;
+  CockpitTransport *transport;
+  CockpitChannel *channel;
+  GBytes *sent;
+
+  mock = mock_transport_new ();
+  transport = COCKPIT_TRANSPORT (mock);
+
+  options = json_object_new ();
+  channel = g_object_new (mock_echo_channel_get_type (),
+                          "transport", transport,
+                          "id", "55",
+                          "options", options,
+                          NULL);
+  cockpit_channel_ready (channel, NULL);
+  json_object_unref (options);
+
+  sent = cockpit_transport_build_control ("command", "ping", "channel", "55", "other", "marmalade", NULL);
+  cockpit_transport_emit_recv (transport, NULL, sent);
+  g_bytes_unref (sent);
+
+  reply = mock_transport_pop_control (mock);
+  g_assert (reply != NULL);
+  cockpit_assert_json_eq (reply, "{ \"command\": \"ready\", \"channel\": \"55\" }");
+
+  reply = mock_transport_pop_control (mock);
+  g_assert (reply != NULL);
+  cockpit_assert_json_eq (reply, "{ \"command\": \"pong\", \"channel\": \"55\", \"other\": \"marmalade\" }");
+
+  g_object_unref (channel);
+  g_object_unref (mock);
+}
+
+static void
+test_ping_no_channel (void)
+{
+  JsonObject *reply = NULL;
+  JsonObject *options;
+  MockTransport *mock;
+  CockpitTransport *transport;
+  CockpitChannel *channel;
+  GBytes *sent;
+
+  mock = mock_transport_new ();
+  transport = COCKPIT_TRANSPORT (mock);
+
+  options = json_object_new ();
+  channel = g_object_new (mock_echo_channel_get_type (),
+                          "transport", transport,
+                          "id", "55",
+                          "options", options,
+                          NULL);
+  json_object_unref (options);
+
+  sent = cockpit_transport_build_control ("command", "ping", "channel", "56", "other", "marmalade", NULL);
```

Comment:
Channel 56 isn't defined, and also the test is called "no channel". Should the channel simply be dropped here? Or did I entirely misunderstand this?

Reply:
It's an unknown channel. I've reworded and added a comment. Also added a line to the docs about this. 

Control messages for unknown channels are ignored in the protocol in general. This is just a further case and test of this.